packages/story-editor/src/components/library/panes/shopping/productButton.js

```diff
@@ -60,41 +64,78 @@ const StyledButton = styled(Button)`
 const StyledActionsContainer = styled.div`
   margin-right: 10px;
 `;
+const TooltipWrapper = ({ hasImage, children }) => {
+  return !hasImage ? (
+    <Tooltip
+      hasTail
+      title={__('Products without images cannot be added.', 'web-stories')}
+    >
+      {children}
+    </Tooltip>
+  ) : (
+    children
+  );
+};
+
+TooltipWrapper.propTypes = {
+  hasImage: PropTypes.bool,
+  children: PropTypes.node,
+};
+
 function ProductButton({ product, onClick, onFocus, isOnPage }) {
   const ADD_PRODUCT_TEXT = sprintf(
     /* translators: %s: product title. */
     __('Add %s', 'web-stories'),
     product?.productTitle
   );
 
+  const hasImage = product?.productImages && product?.productImages?.length;
   const REMOVE_PRODUCT_TEXT = sprintf(
     /* translators: %s: product title. */
     __('Remove %s', 'web-stories'),
     product?.productTitle
   );
 
+  const DISABLED_PRODUCT_TEXT = __(
+    'Products without images cannot be added.',
+    'web-stories'
+  );
+
+  const ariaLabel = () => {
```

Comment:
I suppose it's a preference thing but instead of a function we could use `let ariaLabel = ...` and some if statements. I don't think we typically use functions for that elsewhere.